share/spice/currency/currency.js

```diff
@@ -29,29 +222,150 @@
-        } else {
-            $(".zci--currency-container").css("height", "5em");
-        }
-    }
-    
-    // Change the look of the mobile view if the content overflows.
-    function resizeMobile() {
-        var tileHeight = $(".zci--currency .tile--s").outerHeight();
 
-        if(tileHeight > 155) {
-            $(".zci--currency .tile--s").addClass("large").removeClass("small");
-        } else {
-            $(".zci--currency .tile--s").addClass("small").removeClass("large");
-        }
-    }
-    
+    // Currency Converter
+    var Converter = {
+
+        from_currency: undefined,
+        to_currency: undefined,
+        rate: undefined,
+        inverseRate: undefined,
+
+        //
+        // Retrieves conversion info from JSONP payload
+        //
+
+        getConversionRate: function(payload) {
+            return +$(payload["conversion-rate"].split(" ")).get(-2);
+        },
+
+        getInverseConversionRate: function(payload) {
+            return +$(payload["conversion-inverse"].split(" ")).get(-2);
+        },
+
+        getFromCurrencyName: function(payload) {
+            return payload["from-currency-symbol"];
+        },
+
+        getToCurrencyName: function(payload) {
+            return payload["to-currency-symbol"];
+        },
+
+        getConversionLabel: function(payload) {
+            return payload["conversion-rate"];
+        },
+
+        getInverseConversionLabel: function(payload) {
+            return payload["conversion-inverse"]
+        },
+
+        //
+        // Calculates the rates
+        //
+
+        calculateRate: function() {
+            if($currency_input_left.val() !== '' && /\d+/.test($currency_input_left.val())) {
+                var left_input = $currency_input_left.val()
+                var rightval = parseFloat(left_input) * Converter.rate;
+                $currency_input_right.val(
+                    DDG.commifyNumber(rightval.toFixed(2))
+                );
+            } else {
+                $currency_input_right.val("");
+            }
+        },
+
+        calculateInverseRate: function() {
+            if($currency_input_right.val() !== '' && /\d+/.test($currency_input_right.val())) {
+                var right_input = $currency_input_right.val()
+                var leftval = parseFloat(right_input) * Converter.inverseRate;
+                $currency_input_left.val(
+                    DDG.commifyNumber(leftval.toFixed(2))
+                );
+            } else {
+                $currency_input_left.val("");
+            }
+        },
+
+        getRatesFromAPI: function() {
+
+            var from = $left_select.val();
+            var to = $right_select.val();
+
+            // We'll flip the currencies if the user tries to compare the same symbol
+            if (from === to) {
+                from = Converter.to_currency;
+                to = Converter.from_currency;
+                $right_select.val(to);
```

Comment:
I think you need an analogous line for the left input:
![ddgtemp2](https://cloud.githubusercontent.com/assets/4411471/26224165/2de5aa06-3bef-11e7-8292-be3860c31d1a.gif)